src/lib.rs

```diff
@@ -1,5 +1,102 @@
+use std::marker::PhantomData;
+
 #[cfg(target_os = "macos")]
-pub mod macos;
+mod macos;
+
+#[cfg(target_os = "android")]
+mod android;
+
+#[cfg(target_os = "windows")]
+mod windows;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Unsupported platform")]
+    Unsupported,
+    /// Permissions to access location / device are not allowed, and cannot be accessed until the
+    /// user manually intervenes.
+    #[error("Permissions denied")]
+    PermissionsDenied,
+    #[cfg(target_os = "windows")]
+    #[error("{0}")]
+    Windows(#[from] windows_result::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+pub struct Getter {
+    inner: Inner,
+    _phantom: PhantomData<*mut ()>,
+}
+
+enum Inner {
```

Comment:
is it possible for this enum to have multiple variants at compile time? perhaps for external GPS sources like serial?